Learning git (#4)
mrtrimble/eleventy-starter-project: src/content/docs/github-steps.md

---

> title: Github Steps

Reviewer: ```suggestion
title: GitHub Steps
```

---

## When starting a new project, visit Github to "fork" code.

1. Click “Your Profile” in top right hand corner.
2. Click “Repositories”.
3. Click the repo you’d like to fork.

> 4. Fork in top right hand corner. Create fork.

Reviewer: ```suggestion
1. Browse to the repository you'd like to fork.
2. Click the "Fork" button in the top right corner
3. On the next page, keep the default settings and click "Create Fork" button
```

> 5. Click Code then Copy git url.

Reviewer: ```suggestion
4. While on your forked copy of the repository, click the "Code" button and copy Git URL.
```

> 6. Open “Command Prompt”

Reviewer: ```suggestion
5. Open a Terminal (MacOS) or Command Prompt (Windows)
```

> 7. Type `dir` for a list of folders and press enter.

Reviewer: ```suggestion
6. To list the contents of a directory, enter `ls` (MacOS) or `dir` (Windows)
```

> 8. Type `cd folder the project should go in` then press enter.

Reviewer: ```suggestion
7. Browse into the folder you would like to store the project.
  - You can navigate the directory structure from the command line by using `cd` (change directory). 
```

> 9. Type `git clone https://github.com/vchinnick/eleventy-starter-project.git url`

Reviewer: ```suggestion
8. To clone a repository, enter `git clone` followed by the Git URL of the repository
  - Example: `git clone https://github.com/vchinnick/eleventy-starter-project.git`
```

10. In VS code, click ‘File’ then ‘Open folder’ then choose the folder your project is in.
11. At the top, click ‘Terminal’ then ‘New Terminal’
12. In VS code, create new branch.

>     - `Git checkout -b (name)`

Reviewer: ```suggestion
## Making Changes to a Project

1. In VS code, click "File" then "Open folder" then choose the folder your project is in.
2. At the top, click "Terminal" then "New Terminal"
3. Inside the terminal, create new branch to track your changes.
    - `git checkout -b branch-name`
```

13. To make sure you’re on the right branch, type `git branch`.

> 14. Start adding stuff to the editor. Save when you’re done adding edits.

Reviewer: ```suggestion
4. To verify you’re on the correct working branch, type `git branch` to see your list of branches.
5. Proceed with editing files within the project.
```

15. To see changes, run the project locally [npm start].

> 16. If in Node, switch to powershell.

Reviewer: ```suggestion
## Running a project

1. Different types of projects may have different ways to run a project, always refer to the project `README.md` file to learn how to get up and running on your local machine.
2. If  you are working on a node project, check the `packages.json` file to see a list of scripts available to run within the project.
    - If it is the first time running a node project, make sure to run `npm install` to install all dependencies.
    - Generally `npm start` or `npm run dev` will be available as scripts to start the node server.
```

> 17. Under Source Control, if it’s listed as Untracked–right click and Stage changes OR in terminal use command below..
>     - `git add .` for all files
>     - `git add file-name` for specific files.
> 18. Commit changes to branch
>     - `git commit -am ‘message’`
> 19. Push changes to Github
>     - `git push origin head`
>     - OR use Publish button to publish branch-to origin

Reviewer: ```suggestion
## Committing Changes to a Branch

1. Check for any untracked files
    - You can check the status of your changes by running `git status` inside the terminal.
    - You can also check the status of changes within VS Code's Source Control tab, 
      - Files marked with `M` mean they have been "modified" in the branch.
      - Files marked with `U` mean they are "untracked" in the branch.
      - Files marked with `D` mean they have been "deleted" from the branch.
      - Files marked with `A` mean they have been "added" to the branch.
2. If a file is untracked – it will need to be added to the branch as a tracked file.
    - In the terminal you can add files by entering:
      - `git add file-name` for specific files.
      - `git add .` for all files
    - You can add files through VS Code's Source Control tab as well, by right-clicking the file and choosing "Stage Changes"
3. If there are changes you do not want to be committed to the branch, you can revert individual files or all files before committing.
4. Once all changes have been staged, you can commit changes to the branch running:
    - `git commit -am ‘commit message here’`
5. Push changes to GitHub
    - In the terminal, enter `git push origin head` to push the current working branch to the remote origin repository on GitHub
    - You can also push a branch through VS Code's Source Control tab by pressing the "Publish Branch" button whenever it is available.
```

> 20. Go to github.
> 21. Review changes.
> 22. Create pr.
> 23. Someone will review and merge.
>     - If you already created a PR and you make a change in VS then pushed to Github, you can just click Refresh in github and it will pull the changes automatically in the PR.
>     - If you made changes and don’t want to commit them…go to source control and click Revert button to discard changes.

Reviewer: ```suggestion
## Creating a Pull Request
1. Browse to your repository on GitHub.
2. Look for the yellow box that says "Create Pull Request"
3. Provide a title and description for the Pull Request
4. If available, feel free to tag someone to review the changes in the Pull Request
    - You can continue to push changes to the Pull Request until it is approved and merged.
```

> ## If you already did a project, it’s already in VS code.
> 
> _(I don’t need to fork it, because it’s already in VS code. Forking and cloning is only done once per project.)_
> 
> 1. In VS code, pull in latest changes.
>    - `git fetch upstream`
> 2. Get code up to date.
>    - `git rebase upstream/main`

Reviewer: ```suggestion
## Fetching the Latest Changes to a Project

1. In the terminal, switch back to your master (or main) branch of your cloned repository
    - `git checkout master` or `git checkout main`
2. Fetch the latest changes from the `upstream` branch.
    - `git fetch upstream`
3. Update your local repository by rebasing 
    - `git rebase upstream/master` or `git rebase upstream/main`
```

4. To make sure you’re on the right branch, type ‘git branch’.
5. On the left side, there is a file tree. Select what you want to edit.
6. Content→Docs on Eleventy site.
7. Right click → New File
   - `Name.md`
8. Add Frontmatter-Add three hyphens press enter add three more hyphens. This syntax is for denoting frontmatter beginning and end.
9. Start adding stuff to the editor. Save when you’re done adding edits.
10. To run the project locally , type `npm start`.
11. Click Local URL and it shows changes on site.
12. If in Node, switch to powershell.
13. Under Source Control, if it’s listed as Untracked–right click and Stage changes OR in terminal use command below..
    - `git add .` for all files
    - `git add file-name` for specific files
14. Commit changes to branch
    - `git commit -am ‘message’`
15. Push changes to Github
    - `git push origin head`
    - OR use Publish button to publish branch-to origin
16. Go to github.
17. Review changes.
18. Create pr.
19. Someone will review and merge.
    - If you already created a PR and you make a change in VS then pushed to Github, you can just click Refresh in github and it will pull the changes automatically in the PR.

>     - If you made changes and don’t want to commit them…go to source control and click Revert button to discard changes.

Reviewer: Let's just remove this part!
